### PR TITLE
Removing key from object store when collection object it empty

### PR DESCRIPTION
**Pull request by TalZaccai**
* Fixing bug where keys of emptied collections are not removed from object store (this currently prevents reuse of keys to store object of different type)
* Fixing additional bug in RecordStatus not using the correct mask

### CODE REVIEW

File: libs/server/Objects/List/ListObject.cs [hunk 1]
```diff
@@ -57,6 +57,7 @@ public enum OperationDirection
     /// </summary>
     public partial class ListObject : GarnetObjectBase
     {
+        /// <inheritdoc />
         readonly LinkedList<byte[]> list;
```
Comment (TedHartMS) [inline libs/server/Objects/List/ListObject.cs:61]: Inheritdoc from where? Same question for SetObject

File: libs/server/Objects/SortedSetGeo/SortedSetGeoObjectImpl.cs [hunk 2]
```diff
@@ -15,7 +15,7 @@ namespace Garnet.server
     /// <summary>
     /// Sorted Set - RESP specific operations for GEO Commands
     /// </summary>
-    public unsafe partial class SortedSetObject : GarnetObjectBase
+    public unsafe partial class SortedSetObject : IGarnetObject
     {
```
Comment (TedHartMS) [inline libs/server/Objects/SortedSetGeo/SortedSetGeoObjectImpl.cs:19]: why this change?

File: libs/server/Storage/Session/ObjectStore/Common.cs [hunk 3]
```diff
@@ -30,7 +30,7 @@ unsafe GarnetStatus RMWObjectStoreOperation<TObjectContext>(byte[] key, ArgSlice
             Debug.Assert(_output.spanByteAndMemory.IsSpanByte);
 
 
-            if (!status.Record.Created && !status.Record.CopyUpdated && !status.Record.InPlaceUpdated)
+            if (status.NotFound || (!status.Record.Created && !status.Record.CopyUpdated && !status.Record.InPlaceUpdated))
                 return GarnetStatus.NOTFOUND;
```
Comment (TedHartMS) [inline libs/server/Storage/Session/ObjectStore/Common.cs:34]: We should not need CopyUpdated or InPlaceUpdated to be mentioned here; NotFound cannot be true if they are present. Look at RMW_MainStore (AdvancedOps.cs); it is much more straightforward.

           return status.Found || status.Record.Created ? GarnetStatus.OK : GarnetStatus.NOTFOUND;

File: libs/server/Objects/List/ListObject.cs (hunk 1 above)
Comment (TalZaccai) [reply libs/server/Objects/List/ListObject.cs:61]: Sorry, this was related to something I did before & forgot to remove, deleting these...

File: libs/server/Objects/SortedSetGeo/SortedSetGeoObjectImpl.cs (hunk 2 above)
Comment (TalZaccai) [reply libs/server/Objects/SortedSetGeo/SortedSetGeoObjectImpl.cs:19]: Undid this

**Review by TedHartMS — APPROVED** [review]
